internal/pipeline/executor_test.go

```diff
@@ -0,0 +1,131 @@
+package pipeline
+
+import (
+	"io/ioutil"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/GoogleContainerTools/kpt/internal/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResolveSources(t *testing.T) {
+	// empty directory
+	emptyDir, err := ioutil.TempDir("", "kpt")
+	defer testutil.AssertNoError(t, os.RemoveAll(emptyDir))
+	testutil.AssertNoError(t, err)
+
+	// package with subpackages
+	dir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.FailNow()
+	}
+	defer os.RemoveAll(dir)
+
+	err = ioutil.WriteFile(dir+"/Kptfile", []byte(`
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+`), 0600)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	// first subdirectory of dir
+	subDir1, err := ioutil.TempDir(dir, "")
+	if err != nil {
+		t.FailNow()
+	}
+	defer os.RemoveAll(subDir1)
+
+	err = ioutil.WriteFile(subDir1+"/Kptfile", []byte(`
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+`), 0600)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
+
+	// second subdirectory of dir
+	subDir2, err := ioutil.TempDir(dir, "")
+	if err != nil {
+		t.FailNow()
+	}
+	defer os.RemoveAll(subDir2)
+
+	err = ioutil.WriteFile(subDir2+"/Kptfile", []byte(`
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+`), 0600)
+	if !assert.NoError(t, err) {
+		t.FailNow()
+	}
```

Comment:
Can you pl. check pkgbuilder related test setup @mortent added for `pkg get/update` tests. We need utilities to setup these tests because we need these in other places as well.

Reply:
Acknowledged. I will add a file to testutil that is similar to the current pkgbuilder (since that one has a different pkg struct with different fields)

I think it would make the codebase less confusing if we had different names for these two structs; right now "pkg" is overloaded. WDYT?

Reply:
Acknowledged. I will add a file to testutil that is similar to the current pkgbuilder (since that one has a different pkg struct with different fields)

Reply:
updated the test structure, thanks